libp2p/transports/wstransport.nim

```diff
@@ -331,11 +331,11 @@ proc new*(
   rng: Rng = nil,
   handshakeTimeout = DefaultHeadersTimeout): T =
 
-  T(
-    upgrade: upgrade,
-    tlsPrivateKey: nil,
-    tlsCertificate: nil,
-    flags: flags,
-    factories: @factories,
-    rng: rng,
-    handshakeTimeout: handshakeTimeout)
+  T.new(
```

Comment:
You don't need to call `new` explicitly on a `ref` type, it will always be created on the heap.

Reply:
It's to call the procedure above